crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -483,21 +485,43 @@ impl SlidingSync {
             if self.inner.sticky.read().unwrap().data().extensions.e2ee.enabled == Some(true) {
                 debug!("Sliding Sync is sending the request along with outgoing E2EE requests");
 
-                let (e2ee_uploads, response) =
-                    futures_util::future::join(self.inner.client.send_outgoing_requests(), request)
-                        .await;
+                // If the sliding sync request fails faster than the e2ee requests get their
+                // responses, we want to fail as early as possible, so as to invalidate the
+                // sliding sync session as early as possible.
+                //
+                // Also, we don't want an interruption in the sliding sync request to abort
```

Comment:
Hmm, but aborting on drop will abort processing the e2ee stuff in case of an SS error?

Reply:
Yup, but since it's running concurrently with the SS request, it has a chance to get sent :-p. At worst, the E2EE requests are resent, which should not be a problem, right?

Reply:
I'm updating the comment.